core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -187,19 +193,24 @@ public byte asByte() {
     }
 
     public double asDouble() {
-        if (this == AUTO) {
+        if (isAuto()) {
             return 1d;
         }
         return Double.parseDouble(fuzziness.toString());
     }
 
     public float asFloat() {
-        if (this == AUTO) {
+        if (isAuto()) {
             return 1f;
         }
         return Float.parseFloat(fuzziness.toString());
     }
 
+    // A deserialized fuzziness with AUTO is not the static Fuzziness.AUTO object
+    private boolean isAuto() {
+        return this == AUTO || fuzziness.equalsIgnoreCase("AUTO");
```

Comment:
Maybe my previous comment wasn't clear. I think that if we want to make this class case-insensitive, our current equals impl is wrong too, because "auto" is not equal to "AUTO" but isAuto would return true. I think we should clean this up. Can we simply uppercase the fuzziness string when we parse?